src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Types/TypeManager.cs

```diff
@@ -334,6 +334,12 @@ public PointerType GetPointer(CType baseType)
 
         public NullableType GetNullable(CType pUnderlyingType)
         {
+            if (pUnderlyingType is NullableType nt)
+            {
+                Debug.Fail("Attempt to make nullable of nullable");
+                return nt;
+            }
```

Comment:
This guard is defensive and provides extra assurance that the reasoning that `T??` can't happen is true. Hopefully it can be fully proven as unnecessary and removed.